src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -634,7 +634,13 @@ public function text(string $default = null, bool $normalizeWhitespace = true)
      */
     public function innerText(): string
     {
-        return $this->filterXPath('.//text()')->text();
+        foreach ($this->getNode(0)->childNodes as $childNode) {
+            if (\XML_TEXT_NODE === $childNode->nodeType && '' !== trim($childNode->nodeValue)) {
+                return $this->normalizeWhitespace($childNode->nodeValue);
```

Comment:
Would it make sense to have a flag as we have for `text()`?

Reply:
But then again, it's not consistent with `text`

Reply:
I added it in the first version I pushed, but I thought I should remove it @nicolas-grekas, because you said
> Please remove the new arguments from this PR and keep only the bugfix part.

When it's OK as an optional argument `bool $normalizeWhitespace = true` I can add it again.

Reply:
OK? https://github.com/symfony/symfony/pull/48684/commits/48d39856a6854f8b529879dd0a216044c545fb53